File: pyg-portal-web/src/main/webapp/js/service/specificationService.js
	//绑定一个service
	app.service("specificationService", function($http) {
		//查询所有商品
		this.findAll = function() {
			return $http.get("../specification/findAll");
		};

		//分页查询商品
		this.findPage = function(page, rows) {
			return $http.get("../specification/findPage?page=" + page + "&rows=" + rows);
		};

		//添加
		this.add = function(entity) {
			return $http.post("../specification/add" , entity)
		};
		//修改
		this.update = function(entity) {
			return $http.post("../specification/update" , entity)
		};

		//查询单个商品
		this.findOne = function(id) {
			return $http.get("../specification/findOne?id=" + id);
		}

		//删除商品
		this.dele = function(ids) {
			return $http.get("../specification/delete?ids=" + ids);
		};
		//查询数据,显示select2 下拉列表
		this.findSpecList=function(){
			return $http.get("../specification/findSpecList");
			
		}
		
		

	});
	

File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/SpecificationService.java
package com.pyg.manager.service;

import java.util.List;
import java.util.Map;

import com.pyg.pojo.PageResult;
import com.pyg.pojo.TbBrand;
import com.pyg.utils.PygResult;
import com.pyg.vo.Specification;

public interface SpecificationService {

	// 返回分页列表
	public PageResult findPage(int pageNum, int pageSize);

	// 保存规格--规格选项
	// 参数specification
	public PygResult add(Specification specification);
	
	/**
	 * 根据id查询规格和规格选项 
	 * 参数:id 
	 * 返回值Specification
	 */
	public Specification findOne(Long id);

	//保存规格和规格选项 
	public PygResult update(Specification specification);

	/**
	 * 需求:批量删除规格 参数long数组ids 返回值pygresult
	 */
	public PygResult delete(Long[] ids);

	/**
	 * 查询id和name,用于显示在select2 的下拉列表里
	 */
	public List<Map> findSpecList();

}
